### PR TITLE
Bump Angular to 10.0

### PR DESCRIPTION
**Pull request by oliverguenther**
https://community.openproject.com/wp/33959

### CODE REVIEW

File: frontend/src/app/modules/calendar/wp-calendar/wp-calendar.component.ts [hunk 1]
```diff
@@ -14,18 +14,16 @@ import {DomSanitizer} from "@angular/platform-browser";
...
+import {ToolbarInput} from '@fullcalendar/common';
 import {ConfigurationService} from "core-app/modules/common/config/configuration.service";
 import {UntilDestroyedMixin} from "core-app/helpers/angular/until-destroyed.mixin";
 import {SchemaCacheService} from "core-components/schemas/schema-cache.service";
 
 interface CalendarViewEvent {
   el:HTMLElement;
   event:EventApi;
-  jsEvent:MouseEvent;
 }
```
Comment (openprojectci) [inline frontend/src/app/modules/calendar/wp-calendar/wp-calendar.component.ts:28]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/opf/openproject/blob/f9d0fed92898a03715f25bf0d3d2c524e624b086/frontend/src/app/modules/calendar/wp-calendar/wp-calendar.component.ts#L28"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</del></td>
    </tr>
  </tbody>
</table>

File: frontend/src/app/modules/calendar/wp-calendar/wp-calendar.component.ts [hunk 2]
```diff
@@ -14,18 +23,16 @@ import {DomSanitizer} from "@angular/platform-browser";
...
```
Comment (openprojectci) [inline frontend/src/app/modules/calendar/wp-calendar/wp-calendar.component.ts:37]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/opf/openproject/blob/82a9fab7f62633db5e4ac8701c1184fba140fb3b/frontend/src/app/modules/calendar/wp-calendar/wp-calendar.component.ts#L37"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</del></td>
    </tr>
  </tbody>
</table>

**Comment by openprojectci** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/82a9fab7f62633db5e4ac8701c1184fba140fb3b/frontend/src/app/modules/boards/board/board-filter/board-filter.component.ts#L16">frontend/src/app/modules/boards/board/board-filter/board-filter.component.ts#L16</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by oliverguenther** [comment]
Hi @ulferts , the 10 upgrade part of this PR was straightforward, only a few SASS fixes were added and a list of all the files that use `require(...)`  needed to be added to `angular.json`. 

I included the upgrade to FullCalendar 5, which introduces some styling changes. Could you give the time widget / calendar a spin to see if this works as you'd expect/designed it? Thanks!

**Comment by ulferts** [comment]
@oliverguenther I fixed the styling of both calendars and hope to have restored it completely. 

I did not look at all of the other parts of the application but from what I have seen and what the tests tell me, this should be good to merge, now.

**Pull request merged**